### DISCUSSION TITLE
ClearScript 7.1 Linux support on JavaScript

### DISCUSSION THREAD

**Discussion by cilerler** [ClearFoundry/ClearScript#234] [Q&A] [answered]
> Non-Windows builds are limited to V8; legacy script engines (JScript/VBScript) are non-functional.

I was looking at release notes.  
If the code I'm passing is a JavaScript code, will it continue to work? Or is it fall to the area "JScript is non-functional"
If it will continue to work, what should I worried about?  What shouldn't work?

```csharp
V8ScriptEngine engine = null;
using (engine = new V8ScriptEngine())
{
	engine.AddHostObject("parser", this);
}

engine.Execute(codeStmt.Code);

// ...

var context = new { obj = obj, property = property, value = value };			
engine.AddHostObject("context", context);
string evaluated = engine.Evaluate((string)functionArguments).ToString().Trim();
```

Description of `Microsoft.ClearScript.linux-x64` states 

> A library for adding scripting to .NET applications on Linux (x64). Supports JavaScript via V8.**

not sure what is the difference.

**Comment by ClearScriptLib** [comment, marked as answer]
Hi @cilerler,

Sorry about the lack of clarity in the release notes.

ClearScript on non-Windows platforms fully supports `V8ScriptEngine`. It does _not_ support `JScriptEngine` or `VBScriptEngine`, as those are dependent on native closed-source components that are only available on Windows and long since deprecated. The good news is that `V8ScriptEngine` far exceeds `JScriptEngine` in JavaScript performance and standards support. The bad news is that `VBScriptEngine` is likely to remain Windows-only.

ClearScript on Windows also supports a number of technologies that aren't available on Linux or macOS. These are mostly legacy capabilities such as COM interoperability.

Good luck!
